astropy/table/table.py

```diff
@@ -119,15 +121,28 @@ def __repr__(self):
         names = ("'{0}'".format(x) for x in six.iterkeys(self))
         return "<{1} names=({0})>".format(",".join(names), self.__class__.__name__)
 
-    def _rename_column(self, name, new_name):
+    def _rename_column(self, name, new_name, rename_duplicate=True):
```

Comment:
I think this `_rename_column` private method can just stay entirely unchanged, because it should definitely have a default of `rename_duplicate=False`.  I don't want to allow this behavior by default because it can let program bugs go by unnoticed (even with warnings).

Since `_rename_column` is only called in the column name setter, which doesn't allow overriding the default, that means it isn't possible to call this with `rename_duplicate=True`.  So that makes it easy - this bit can be unchanged.

(Hopefully I didn't miss something in the logic there..)